armi/reactor/components/componentParameters.py

```diff
@@ -63,6 +64,35 @@ def getComponentParameterDefinitions():
             description="Number densities of each nuclide.",
         )
 
+        pb.defParam(
+            "detailedNDens",
+            setter=isNumpyArray("detailedNDens"),
+            units=f"atoms/(bn*{units.CM})",
+            description=(
+                "High-fidelity number density vector with up to thousands of nuclides. "
+                "Used in high-fi depletion runs where low-fi depletion may also be occurring. "
+                "This param keeps the hi-fi and low-fi depletion values from interfering. "
+                "See core.p.detailedNucKeys for keys. "
+                # Could be moved to external physics plugin
+            ),
+            saveToDB=True,
+            default=None,
+        )
+
+        pb.defParam(
+            "powerDecay",
```

Comment:
what is the purpose of adding the `powerDecay` and `pdensDecay` params?

Reply:
These are for calculating decay heat. Since we moved detailed depletion to the component level to support axial expansion, we now need these params at the component level.